hazelcast/src/main/java/com/hazelcast/internal/partition/operation/AbstractPartitionOperation.java

```diff
@@ -70,7 +71,11 @@ final Collection<Operation> createNonFragmentedReplicationOperations(PartitionRe
             UrgentPartitionRunnable<Collection<Operation>> runnable = new UrgentPartitionRunnable<>(
                     event.getPartitionId(), () -> createReplicationOperations(event, true));
             getNodeEngine().getOperationService().execute(runnable);
-            return runnable.future.joinInternal();
+            try {
+                return runnable.future.get();
```

Comment:
this changes the type of exceptions thrown: `get()` wraps in `ExecutionException`, `joinInternal` throws the exception itself (if a `RuntimeException`) or wraps in a `HazelcastException`. Why would we need this change?

Reply:
Why do we replace `joinInternal()` with a `get()` method?
Because `join()` and `joinInternal()` are not interrupted in a case of `ExcecutorService#shutdownNow()`

Reply:
Maybe we could wrap `get()` exceptions in a `HazelcastException`

Reply:
> Because join() and joinInternal() are not interrupted

right, forgot about that.

We should however unwrap/wrap accordingly, so that we don't change throwing conventions. (Maybe `rethrow` or `peel` in `ExceptionUtil` does the trick?)

Reply:
Yes, I'll check. Thanks for the comment!